Source/controls/game_controls.cpp

```diff
@@ -56,50 +69,72 @@ SDL_Keycode TranslateControllerButtonToKey(ControllerButton controllerButton)
 	}
 }
 
-bool HandleStartAndSelect(const ControllerButtonEvent &ctrlEvent, GameAction *action)
+SDL_Keycode TranslateControllerButtonToQuestLogKey(ControllerButton controllerButton)
 {
-	const bool inGameMenu = InGameMenu();
-
-	const bool startIsDown = IsControllerButtonPressed(ControllerButton_BUTTON_START);
-	const bool selectIsDown = IsControllerButtonPressed(ControllerButton_BUTTON_BACK);
-	start_modifier_active = !inGameMenu && startIsDown;
-	select_modifier_active = !inGameMenu && selectIsDown && !start_modifier_active;
-
-	// Tracks whether we've received both START and SELECT down events.
-	//
-	// Using `IsControllerButtonPressed()` for this would be incorrect.
-	// If both buttons are pressed simultaneously, SDL sends 2 events for which both buttons are in the pressed state.
-	// This allows us to avoid triggering START+SELECT action twice in this case.

```

Comment:
Does the new code handle this case correctly?

Reply:
I'm not sure this case matters much. Unless I misunderstood something here, it's trying to handle cases like when the player is holding `Select` and then accidentally presses `Start` and `A` at the exact same time. This code would swallow the `A` input and only trigger `Select+Start`. The padmapper would depend on what order SDL sends the events, but the case is highly unlikely to happen, and it shouldn't be at all surprising to the user if we handle the extra button.

Maybe I don't understand the use-case, though. I suppose there could be a device that sends spurious `A` events when `Start` and `Select` are pressed together or something..?

Reply:
The case handled here is actually just pressing SELECT + START simultaneously. If you can, please give it try it with SDL2.
Pressing SELECT + START triggers 2 events: the first opened the menu and the second immediately closed it in the previous implementation before this workaround.

Reply:
Actually, I needed to reread the whole comment. If `Start+Select` are pressed simultaneously, it would trigger two events and the game menu would open and immediately close because the second event would be translated to the Escape key. I imagine this could happen more often on devices that aren't processing events at 150+ fps.

I was initially thinking this would be handled properly because the `Start` event would come in and trigger its own action, thereby becoming the modifier key for `Select`. However, when I eventually implemented `PadmapperOptions::ButtonPressed()`, I ended up using `IsControllerButtonPressed()` and falling into the same trap. I'll try to think of a way to iron this out.

Reply:
This should be resolved. Refer to the logic related to the `SuppressedButton` variable.